rustsec/src/database.rs

```diff
@@ -47,22 +47,21 @@ impl Database {
 
         for collection in Collection::all() {
             let collection_path = path.join(collection.as_str());
+            let collection_entry = fs::read_dir(&collection_path)?;
 
-            if let Ok(collection_entry) = fs::read_dir(&collection_path) {
-                for dir_entry in collection_entry {
-                    let dir_entry = dir_entry?;
-                    if !dir_entry.file_type()?.is_dir() {
+            for dir_entry in collection_entry {
```

Comment:
Is there some scenario where we should not error if the `Crates` entry exists but the `Rust` entry does not? It seems reasonable to me that conceptually, `Database::open()` should get you both kinds of advisories and anything else is a bit dangerous in potentially failing open.

Reply:
Updated to avoid raising an error for the `Rust` collection.